Init menus

mattcollier opened the pull request.

mattcollier committed:
Implement menu item visibility option.

dlongley commented on navbar.html:
```diff
@@ -42,7 +42,11 @@
         <div br-slot="br-navbar-toolbar">
           <div class="navbar-right">
-            <div br-slot="br-navbar-tool"></div>
+            <div class="btn-toolbar">
```
Does this force tools to be buttons? Perhaps we want the toolbar to have  a slot for tools and a built-in button toolbar with a different slot for each toolbar-button?

mattcollier replied:
Here is the css for the btn-toolbar class: https://github.com/digitalbazaar/bedrock-angular/blob/master/css/app.css#L3329

dlongley replied:
It's more about the semantics of it. Bootstrap probably expects only buttons to be in a btn-toolbar. That tells me that we should have two slots here ... one for entirely custom navbar-tools (br-navbar-tool) and then a built-in, by default empty, button toolbar that has a "br-navbar-toolbar-button" slot.

mattcollier commented:
@dlongley recommendations implemented and tested with the login button and the user profile info in use in bedrock-idp.

dlongley committed:
Remove unnecessary check.

dlongley committed:
Call `menu.init` using `menu` as context.

dlongley committed:
Restore ability to override navbar toolbar.

dlongley committed:
Remove `btn-group`.

- Buttons on the toolbar may not necessarily be tightly grouped
  together into a single unit; allow content that fills in the
  slots make the decision to use a `btn-group` as they see fit.

dlongley committed:
Rename slot to `br-navbar-button-toolbar`.

- Naming better reflects that buttons are appended to a button
  toolbar, not to another button.
